api/repositories/service_offering_repository_test.go

```diff
@@ -448,4 +450,118 @@ var _ = Describe("ServiceOfferingRepo", func() {
+			offeringName := uuid.NewString()
+			broker = &korifiv1alpha1.CFServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: rootNamespace,
+					Name:      brokerGUID,
+					Labels: map[string]string{
+						korifiv1alpha1.GUIDLabelKey: brokerGUID,
+					},
+				},
+				Spec: korifiv1alpha1.CFServiceBrokerSpec{
+					Name: offeringName,
+				},
+			}
+			Expect(k8sClient.Create(ctx, broker)).To(Succeed())
+
+			metadata, err := korifiv1alpha1.AsRawExtension(map[string]any{
+				"offering-md": "offering-md-value",
+			})
+
+			Expect(err).NotTo(HaveOccurred())
+			serviceOffering = &korifiv1alpha1.CFServiceOffering{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: rootNamespace,
+					Name:      offeringGUID,
+					Labels: map[string]string{
+						korifiv1alpha1.RelServiceBrokerGUIDLabel: broker.Name,
+						korifiv1alpha1.RelServiceBrokerNameLabel: tools.EncodeValueToSha224(broker.Spec.Name),
+						korifiv1alpha1.CFServiceOfferingNameKey:  tools.EncodeValueToSha224(offeringName),
+						korifiv1alpha1.GUIDLabelKey:              offeringGUID,
+					},
+					Annotations: map[string]string{
+						"annotation": "annotation-value",
+					},
+				},
+				Spec: korifiv1alpha1.CFServiceOfferingSpec{
+					Name:             "my-offering",
+					Description:      "my offering description",
+					Tags:             []string{"t1"},
+					Requires:         []string{"r1"},
+					DocumentationURL: tools.PtrTo("https://my.offering.com"),
+					BrokerCatalog: korifiv1alpha1.ServiceBrokerCatalog{
+						ID:       "offering-catalog-guid",
+						Metadata: metadata,
+						Features: korifiv1alpha1.BrokerCatalogFeatures{
+							PlanUpdateable:       true,
+							Bindable:             true,
+							InstancesRetrievable: true,
+							BindingsRetrievable:  true,
+							AllowContextUpdates:  true,
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, serviceOffering)).To(Succeed())
+
+			updateMessage = repositories.UpdateServiceOfferingMessage{
+				GUID: offeringGUID,
+				MetadataPatch: repositories.MetadataPatch{
+					Labels: map[string]*string{
+						"new-offering-label": tools.PtrTo("new-offering-label-value"),
+					},
+					Annotations: map[string]*string{
+						"new-offering-annotation": tools.PtrTo("new-offering-annotation-value"),
+					},
+				},
+			}
+		})
+
+		JustBeforeEach(func() {
+			updatedServiceOffering, updateErr = repo.UpdateServiceOffering(ctx, authInfo, updateMessage)
+		})
+
+		It("fails because the user has no offerings", func() {
+			Expect(updateErr).To(matchers.WrapErrorAssignableToTypeOf(apierrors.ForbiddenError{}))
+		})
+
+		When("the user is a CFAdmin", func() {
+			BeforeEach(func() {
+				createRoleBinding(ctx, userName, adminRole.Name, rootNamespace)
+			})
+
+			It("updates the service offering metadata", func() {
+				Expect(updateErr).NotTo(HaveOccurred())
+
+				Expect(updatedServiceOffering.Metadata.Labels).To(HaveKeyWithValue(korifiv1alpha1.RelServiceBrokerGUIDLabel, broker.Name))
```

Comment:
You could combine multiple assertions on the same map via `SatisfyAll`, for example:

```golang
Expect(updatedServiceOffering.Metadata.Labels).To(SatisfyAll(
  HaveKeyWithValue(....),
  HaveKeyWithValue(....),
))
```

Alternatively, you could use `gstruct.MatchKeys`, i.e.

```golang
Expect(updatedServiceOffering.Metadata.Labels).To(MatchKeys(IgnoreExtras, Keys{
  "k1": Equal("v1"),
  "k2": Equal("v2"),
}))
```

Reply:
Cool. Updated test.